### PR TITLE
[DOC] Organization Updates

### PR DESCRIPTION
**Pull request by krisdale**
Organizational changes:
- move api-mapping-status files into reference folder
- move diagnostic ref files into reference folder
- move term mapping quick ref to reference folder
- move compare prog models to reference folder
- rename 'migration support status' to 'migration support'
- rename 'migrate a project' folder to 'migration'

Addition of option functional groups (no changes to content)

Refactor 'Migrate a project on Linux', ;Migrate a project on Windows', and 'Before you begin' into new 'Tool setup and basic use' page.

Minor content refactoring to support the organizational changes.

### CODE REVIEW

File: docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+The following table lists |tool_name| command line options that are deprecated.
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Deprecated Option
+     - Suggested Replacement
+   * - .. include:: /_include_files/options_def.rst
+          :start-after: opt-custom-helper-name:
```
Comment (zhimingwang36) [inline docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst]: This option -custom-helper-name have been removed.

File: docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          :end-before: desc-custom-helper-name:
+     - No replacement
+   * - .. include:: /_include_files/options_def.rst
+          :start-after: opt-no-cl-namespace-inline:
```
Comment (zhimingwang36) [inline docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst]: --no-cl-namespace-inline has been already removed in the latest code.

File: docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          :end-before: desc-no-cl-namespace-inline:
+     - .. include:: /_include_files/options_def.rst
+          :start-after: opt-use-explicit-namespace:
+          :end-before: desc-use-explicit-namespace:
+   * - .. include:: /_include_files/options_def.rst
+          :start-after: opt-use-custom-helper:
```
Comment (zhimingwang36) [inline docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst]: -use-custom-helper:  have been removed already.

File: docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst (hunk 1 above)
Comment (krisdale) [reply docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst]: Done.

File: docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst (hunk 2 above)
Comment (krisdale) [reply docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst]: Done.

File: docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst (hunk 3 above)
Comment (krisdale) [reply docs/dev_guide/reference/command-line-options-ref/deprecated-options.rst]: Done.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**